.buildkite/custom-tests.json

```diff
@@ -0,0 +1,82 @@
+{
+  "tests": [
+    {
+      "test_name": "staging: build-gnu",
+      "command": "cd staging && RUSTFLAGS=\"-D warnings\" cargo build --release",
+      "soft_fail": "true",
+      "platform": [
+        "x86_64",
+        "aarch64"
+      ]
+    },
+    {
+      "test_name": "staging: build-musl",
+      "command": "cd staging && RUSTFLAGS=\"-D warnings\" cargo build --release --target {target_platform}-unknown-linux-musl",
+      "soft_fail": "true",
+      "platform": [
+        "x86_64",
+        "aarch64"
+      ]
+    },
+    {
+      "test_name": "staging: style",
+      "command": "cd staging && cargo fmt --all -- --check --config format_code_in_doc_comments=true",
+      "soft_fail": "true"
+    },
+    {
+      "test_name": "staging: unittests-gnu",
+      "command": "cd staging && cargo test --all-features --workspace",
+      "soft_fail": "true",
+      "platform": [
+        "x86_64",
+        "aarch64"
+      ]
+    },
+    {
+      "test_name": "staging: unittests-musl",
+      "command": "cd staging && cargo test --all-features --workspace --target {target_platform}-unknown-linux-musl",
+      "soft_fail": "true",
+      "platform": [
+        "x86_64",
+        "aarch64"
+      ]
+    },
+    {
+      "test_name": "staging: clippy",
+      "command": "cd staging && cargo clippy --workspace --bins --examples --benches --all-features --all-targets -- -D warnings -D clippy::undocumented_unsafe_blocks",
+      "soft_fail": "true",
```

Comment:
Here too.

Reply:
The only thing that remains, is for example what's happening now. The crate in staging doesn't compile because we don't have the dependencies in the container. Is that acceptable?

Reply:
> If we skip here, I think we can skip everywhere, since this will fail if `cargo build` fails.

Sure ? I saw clippy passing here and builds failing..

https://buildkite.com/rust-vmm/vhost-device-ci/builds/1792

> After a second thought, I think maybe it is better as you say and don't skip anything.

You can't go back once I agree with you :)

Reply:
> The only thing that remains, is for example what's happening now. The crate in staging doesn't compile because we don't have the dependencies in the container. Is that acceptable?

I think we broke many exceptions for Sound and I would not like to do it going forward. The tooling needs to be available first and only then we can do all these things.

Reply:
> > If we skip here, I think we can skip everywhere, since this will fail if `cargo build` fails.
> 
> Sure ? I saw clippy passing here and builds failing..
> 
> https://buildkite.com/rust-vmm/vhost-device-ci/builds/1792

mmm, I saw all `staging: *` failing except for `staging: style` where I already removed the `soft_fail`

> 
> > After a second thought, I think maybe it is better as you say and don't skip anything.
> 
> You can't go back once I agree with you :)

ahahah, why not ;-P

Reply:
> mmm, I see all `staging: *` failing except for `staging: style` where I already removed the `soft_fail`

Sorry about that. I saw the non-staging ones by mistake. So okay, just remove the `soft_fail` for `style` and we are good for now.

Reply:
ack, I pushed that change, so the current status should be fine ;-)